build/chip/java/config.gni

```diff
@@ -15,6 +15,7 @@
 java_path = getenv("JAVA_PATH")
 declare_args() {
   java_matter_controller_dependent_paths = []
+  matter_enable_java_generated_api = true
```

Comment:
Could you add a comment describing this?

I wonder if having `type_safe_api` or similar in the name would be more descriptive, as we still seem to include some generated files: the Attribute and Event Value Decoders are there.

Reply:
As you said, I haven't included some decode/encode files in this work yet.
This is excluded because it is used by Android ChipTool, and I think it should be done as a separate task after the cleanup is completed.
(https://github.com/project-chip/connectedhomeip/issues/27398)

Comments will be added inside the source code.

Reply:
I am thinking to disable encode/decoder for events and attributes from cluster objects in controller java/jni layer, actually I feel no one use that right now, I would imagine sdk would plumber Json or tlv blob into application, application would have the freedom and knowledge for cluster

Reply:
@yunhanw-google 
There is a part where I use some decoding / encoding code, so I think this part should be separated.

Since this feature deletes all generated APIs, I will modify it to include all decoding/encoding codes, and then think about how to create a separate library for encoding/decoding codes.

Reply:
@andy31415 
I added comments in source code.

@yunhanw-google 
I disabled tlv decoder. please review this code.